src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,370 @@
+---
+title: Introduction to htmx
+authorName: Niraj Acharya
+authorAvatar: https://avatars.githubusercontent.com/u/122071597?v=4
+authorLink: https://github.com/nirajacharya2
+createdAt: Apr 3, 2024
+tags: js, html, go, htmx
+banner: https://raw.githubusercontent.com/bigskysoftware/htmx/master/www/static/img/htmx_logo.1.png
+---
+
+htmx is an UI library written in JavaScript that extends the features of HTML. It can be extended to make AJAX requests, do CSS transitions, connect to WebSockets and handle Server Sent Events inside a HTML document with just some declarative HTML attributes.
+
+htmx was created by Carson Gross in 2013 because he found HTML to be limited for what he was trying to achieve. htmx tries to solve the following limitations of HTML:
+
+- only `<a>` and `<form>` tags are able to make HTTP requests
+- only `click` & `submit` events trigger them
+- only `GET` & `POST` methods are available
+- and when using `<a>` and `<form>` the whole screen gets replaced
+
+In this blog, we will learn how htmx enhances HTML and overcomes the limitations mentioned above.
+
+**Note:**
+
+I will be using go for the code examples given in this blog post.
+If you are not familiar with go you can learn the basics from [here](https://blog.jankaritech.com/#/blog/Journey%20With%20Go%20-%20A%20Blog%20Series/Go%20Basics%20-%20The%20Starting%20Point).
+
+Through out this blog I will be providing small snippets of my code instead of the actual implementation and I will not go into how to create a webserver to serve the endpoints. For that you might want to read [this blog](https://blog.jankaritech.com/#/blog/Serving%20your%20cat%20video%20using%20the%20Internet%20-%20A%20basic%20understanding%20of%20HTTP%20and%20socket%20programming%20(in%20Java))
+
+
+## Using htmx
+
+We can simply use htmx by adding a script tag in the header of our HTML.
+
+```html
+<head>
+  <script src="https://unpkg.com/htmx.org@1.9.5"></script>
+</head>
+```
+
+htmx claims to be small (~14k min.gz’d), dependency-free, extendable, IE11 compatible & 67% less code when compared with react.
+
+Now let's see how to make a simple AJAX request in htmx.
+
+## Making AJAX request (GET)
+
+First, let's make an endpoint to get data from. I will be using go for the code examples given in this blog post.
+If you are not familiar with go you can learn the basics from [here](https://blog.jankaritech.com/#/blog/Journey%20With%20Go%20-%20A%20Blog%20Series/Go%20Basics%20-%20The%20Starting%20Point).
+
+So, let's begin by creating few endpoints for this example.
```

Comment:
having moved the general notes into the introduction, you can merge line 46 and line 49 into one sentence or delete one of them since they say more or less the same.